lib/kube/utils/utils.go

```diff
@@ -163,9 +163,36 @@ func KubeClusterNames(ctx context.Context, p KubeServicesPresence) ([]string, er
 	return extractAndSortKubeClusterNames(kss), nil
 }
 
-// ListKubeClusterNamesWithFilters returns a sorted list of unique kubernetes clusters
+func extractAndSortKubeClusterNames(kss []types.Server) []string {
```

Comment:
Can't we make this call `extractAndSortKubeClusters` and then just add the names to a list? That way we don't need to maintain effectively the same logic in two places.